### PR TITLE
Remove default option for batch scheduler name

### PR DESCRIPTION
**Pull request by yangwwei**
Per review comment here: https://github.com/ray-project/kuberay/pull/2300#issuecomment-2337070070. Remove the redundant "default" option for the batch scheduler name.

How this was tested?

- UT (this PR added UT coverage for various cases)
- Local tests

```
// normal start up
./ray-operator/bin/manager -leader-election-namespace default -use-kubernetes-proxy

// batchscheduler.enabled=true
// this enables volcano integration
./ray-operator/bin/manager -enable-batch-scheduler -leader-election-namespace default -use-kubernetes-proxy

// batchscheduler.name=volcao
// this enables volcano integration
./ray-operator/bin/manager -batch-scheduler volcano -leader-election-namespace default -use-kubernetes-proxy
{"level":"info","ts":"2024-09-10T16:18:31.104-0700","logger":"setup","msg":"Feature flag batch-scheduler is enabled","scheduler name":"volcano"}
...

// batchscheduler.name=yunikorn
// this enables yunikorn integration
./ray-operator/bin/manager -batch-scheduler yunikorn -leader-election-namespace default -use-kubernetes-proxy
{"level":"info","ts":"2024-09-10T16:18:54.893-0700","logger":"setup","msg":"Feature flag batch-scheduler is enabled","scheduler name":"yunikorn"}
...
```

### CODE REVIEW

**Comment by yangwwei** [comment]
@andrewsykim @kevin85421 pls help to review this follow up PR.

File: ray-operator/main.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func main() {
 		"Encoder to use for logging stdout. Valid values are 'json' and 'console'. Defaults to 'json'")
 	flag.BoolVar(&enableBatchScheduler, "enable-batch-scheduler", false,
 		"(Deprecated) Enable batch scheduler. Currently is volcano, which supports gang scheduler policy.")
-	flag.StringVar(&batchScheduler, "batch-scheduler", "default",
+	flag.StringVar(&batchScheduler, "batch-scheduler", "",
 		"Batch scheduler name, supported values are default, volcano, yunikorn.")
```
Comment (kevin85421) [inline ray-operator/main.go:96]:
```suggestion
		"Batch scheduler name, supported values are volcano and yunikorn.")
```

File: ray-operator/main.go [hunk 2]
```diff
@@ -92,7 +92,7 @@ func main() {
 		"Encoder to use for logging stdout. Valid values are 'json' and 'console'. Defaults to 'json'")
 	flag.BoolVar(&enableBatchScheduler, "enable-batch-scheduler", false,
 		"(Deprecated) Enable batch scheduler. Currently is volcano, which supports gang scheduler policy.")
```
Comment (andrewsykim) [inline ray-operator/main.go:94]: Can we add a note in this flag "Use --batch-scheduler instead"

File: helm-chart/kuberay-operator/values.yaml [hunk 3]
```diff
@@ -78,10 +78,9 @@ batchScheduler:
   # Deprecated. This option will be removed in the future.
   # Note, for backwards compatibility. When it sets to true, it enables volcano scheduler integration.
   enabled: false
-  # Name of the scheduler, currently supported "default", "volcano" and "yunikorn",
-  # set the customized scheduler name, e.g "volcano" or "yunikorn", do not set
+  # Set the customized scheduler name, supported values are "volcano" or "yunikorn", do not set
   # "batchScheduler.enabled=true" at the same time as it will override this option.
-  name: default
+  name: ""
```
Comment (andrewsykim) [inline helm-chart/kuberay-operator/values.yaml:83]: Should we update `helm-chart/kuberay-operator/templates/deployment.yaml` such that --batch-scheduler flag is not set if name is empty?

Comment (yangwwei) [reply helm-chart/kuberay-operator/values.yaml:83]: It is already there: https://github.com/ray-project/kuberay/blob/0e1c248eaab9176251c14d5dbae5011d57e5c154/helm-chart/kuberay-operator/templates/deployment.yaml#L60-L67.

Comment (yangwwei) [reply helm-chart/kuberay-operator/values.yaml:83]: Also did some tests:

1. Default install
```
helm install kuberay-operator ../helm-chart/kuberay-operator
```
the args:
```
   Args:
      --feature-gates=RayClusterStatusConditions=false
      --enable-leader-election=true
```

2. Legacy option
```
helm install kuberay-operator --set batchScheduler.enabled=true ../helm-chart/kuberay-operator 
```
the args in the deployment:
```
    Args:
      --feature-gates=RayClusterStatusConditions=false
      --enable-batch-scheduler
      --enable-leader-election=true
```

3. New option
```
helm install kuberay-operator  --set batchScheduler.name=yunikorn ../helm-chart/kuberay-operator
```
the args

```
   Args:
      --feature-gates=RayClusterStatusConditions=false
      --batch-scheduler=yunikor
      --enable-leader-election=true
```

File: ray-operator/main.go (hunk 1 above)
Comment (yangwwei) [reply ray-operator/main.go]: Thanks. Fixed.

File: ray-operator/main.go (hunk 2 above)
Comment (yangwwei) [reply ray-operator/main.go]: Thanks. Fixed.

**Comment by yangwwei** [comment]
Not sure what are the errors in the CI mean:
> Error: This request has been automatically failed because it uses a deprecated version of `actions/download-artifact: v2`. Learn more: https://github.blog/changelog/2024-02-13-deprecation-notice-v1-and-v2-of-the-artifact-actions/

**Comment by kevin85421** [comment]
> Seems like all PR recently failed with this.

@yangwwei could you rebase with the master branch? Thanks!

**Review by andrewsykim — COMMENTED** [review]
LGTM

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**